modules/samples/firealarm/component/plugin/src/main/java/org/homeautomation/firealarm/plugin/impl/DeviceTypeManager.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.homeautomation.firealarm.plugin.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.homeautomation.firealarm.plugin.exception.FirealarmPluginException;
+import org.homeautomation.firealarm.plugin.impl.dao.DeviceTypeDAO;
+import org.wso2.carbon.device.mgt.common.Device;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.DeviceManager;
+import org.wso2.carbon.device.mgt.common.EnrolmentInfo;
+import org.wso2.carbon.device.mgt.common.FeatureManager;
+import org.wso2.carbon.device.mgt.common.configuration.mgt.TenantConfiguration;
+import org.wso2.carbon.device.mgt.common.license.mgt.License;
+import org.wso2.carbon.device.mgt.common.license.mgt.LicenseManagementException;
+
+import java.util.List;
+
+/**
+ *
+ */
+public class DeviceTypeManager implements DeviceManager {
+
+    private static final Log log = LogFactory.getLog(DeviceTypeManager.class);
+
+    private static final DeviceTypeDAO connectedCupDAO = new DeviceTypeDAO();
+
+    @Override
+    public FeatureManager getFeatureManager() {
+        return null;
+    }
+
+    @Override
+    public boolean saveConfiguration(TenantConfiguration tenantConfiguration)
+            throws DeviceManagementException {
+        //TODO implement this
+        return false;
+    }
+
+    @Override
+    public TenantConfiguration getConfiguration() throws DeviceManagementException {
+        //TODO implement this
+        return null;
+    }
+
+    @Override
+    public boolean enrollDevice(Device device) throws DeviceManagementException {
+        boolean status;
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Enrolling a new Connected Cup device : " + device.getDeviceIdentifier());
+            }
+            DeviceTypeDAO.beginTransaction();
+            status = connectedCupDAO.getConnectedCupDeviceDAO().addDevice(device);
+            DeviceTypeDAO.commitTransaction();
+        } catch (FirealarmPluginException e) {
```

Comment:
How can a FirealarmPluginException occur when adding a device of type "connectedCup"?